mitodl/mit-learn
# drawer v2 - share resource

### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/5876

### Description (What does it do?)
This PR adds the "share" section of the new learning resource drawer and adjusts the layout of the call to section. 

### Screenshots (if appropriate):
![image](https://github.com/user-attachments/assets/15bf3a17-0961-40fc-822f-e598122e4b19)
![image](https://github.com/user-attachments/assets/7d6bb561-95d8-4ae2-bd2d-7aa858965f1b)

### How can this be tested?
 - If you have Posthog set up locally, enable the `lr_drawer_v2` flag
 - If you don't have Posthog set up locally, you can force `drawerV2` to be `true` in `LearningResourceDrawer.tsx`
 - Spin up this branch of `mit-learn`
 - Ensure you have sufficient data backpopulated into your local instance to test a variety of resources
 - Visit http://localhost:8062/search
 - Do *not* log in yet
 - Click any of the search results to bring up the learning resource drawer
 - Verify that you see two buttons below the unit logo, Bookmark and Share
 - Click the Share button
 - Verify that the share section appears and that the URL in the textbox looks correct
 - Click the social media share icons and verify that you are taken to each given social media site and their respective interfaces are displayed for sharing a post with a link to the resource
 - Click the Copy Link button and verify that the link is properly copied to your clipboard
 - Log in as a superuser
 - Go back to the search page and click on any learning resource to bring the drawer back up
 - Verify that you now see 3 buttons; Add to list, Bookmark and Share
